Common/Retranslator/src/Retranslator.cpp

```diff
@@ -0,0 +1,156 @@
+#include "Retranslator.hpp"
+#include <typeinfo>
+
+using namespace AST;
+
+std::unique_ptr<Retranslator> Retranslator::instance = nullptr;
+
+ASTTree Retranslator::parseTree(std::map<std::string, Shape*> map){
+    
+    Node* root = new Node("root");
+    root->addChild(new Node("STARTGRAPH"));
+
+    for (const auto& elem : map)
+    {
+        root->addChild(new Node("STATEMENT"));
+        // if (dynamic_cast<Note*>(elem.second))
+        // {
+        //     root->addChild(makeNote(elem));
+        // }
+        if (dynamic_cast<Line*>(elem.second))
+        {
+            root->addChild(makeLink(elem));
+        }
+        else 
+        {
+            root->addChild(makeObject(elem));
+        }
+    }
+
+    root->addChild(new Node("ENDGRAPH"));
+
+    return ASTTree(root);
+}
+
+Node* doubleNode(std::string key, std::string value, bool isText){
```

Comment:
```suggestion
Node* doubleNode(std::string const& key, std::string const& value, bool isText){
```
Название функции не о чём не говорит, нужно переименновать, чтобы отражало суть